data/data/azure/bootstrap/main.tf

```diff
@@ -152,6 +152,7 @@ resource "azurerm_virtual_machine" "bootstrap" {
 }
 
 resource "azurerm_network_security_rule" "bootstrap_ssh_in" {
+  count                       = var.preexisting_network ? 0 : 1
```

Comment:
In Azure, only one network security group can be attached to a vnet. Since we are depending on an existing vnet, we can't create network security groups or network security rules. So if a vnet exists already, we condition this rule (and others) to not be created.